swarm/storage/resource.go

```diff
@@ -262,7 +262,7 @@ func (self *ResourceHandler) SetStore(store *NetStore) {
 func (self *ResourceHandler) Validate(key Key, data []byte) bool {
 	signature, period, version, name, parseddata, _, err := self.parseUpdate(data)
 	if err != nil {
-		if len(data) == indexSize {
+		if len(data) > indexSize {
```

Comment:
why 🤔

Reply:
also, would you be kind enough to document what is `indexSize`? or rather, find another name for it?
the name is confusing - is it a size of a header? size of data packet? a size of an index inside a byte slice? is it a size of an indice? does it refer to content length or to a step-size?

Reply:
I've added comment by the declaration/initializer, an explanation in the long comment for resourcehandler, and also a comment explaining the need for gt.

Sorry for being cryptic.